core/kazoo_amqp/src/api/kapi_notifications.erl

```diff
@@ -12,486 +12,274 @@
-                                 ,<<"Port-Request-ID">>
-                                 ,<<"Reason">>
-                                      | ?DEFAULT_OPTIONAL_HEADERS
-                                 ]).
--define(PORTED_VALUES, [{<<"Event-Category">>, <<"notification">>}
-                       ,{<<"Event-Name">>, <<"ported">>}
-                       ]).
--define(PORTED_TYPES, [{<<"Reason">>, fun kz_json:is_json_object/1}]).
-
-%% Notify Ported Request
--define(PORT_COMMENT_HEADERS, [<<"Account-ID">>, <<"Comment">>]).
--define(OPTIONAL_PORT_COMMENT_HEADERS, [<<"Authorized-By">>
-                                       ,<<"Local-Number">>
-                                       ,<<"Number">>
-                                       ,<<"Number-State">>
-                                       ,<<"Port">>
-                                       ,<<"Port-Request-ID">>
-                                            | ?DEFAULT_OPTIONAL_HEADERS
-                                       ]).
--define(PORT_COMMENT_VALUES, [{<<"Event-Category">>, <<"notification">>}
-                             ,{<<"Event-Name">>, <<"port_comment">>}
-                             ]).
--define(PORT_COMMENT_TYPES, []).
-
-%% Notify Cnam Request
--define(CNAM_REQUEST_HEADERS, [<<"Account-ID">>, <<"Number">>, <<"Cnam">>]).
--define(OPTIONAL_CNAM_REQUEST_HEADERS, [<<"Acquired-For">>
-                                       ,<<"Local-Number">>
-                                       ,<<"Number-State">>
-                                       ,<<"Request">>
-                                            | ?DEFAULT_OPTIONAL_HEADERS
-                                       ]).
--define(CNAM_REQUEST_VALUES, [{<<"Event-Category">>, <<"notification">>}
-                             ,{<<"Event-Name">>, <<"cnam_request">>}
-                             ]).
--define(CNAM_REQUEST_TYPES, []).
 
-%% Notify Low Balance
--define(LOW_BALANCE_HEADERS, [<<"Account-ID">>, <<"Current-Balance">>]).
--define(OPTIONAL_LOW_BALANCE_HEADERS, ?DEFAULT_OPTIONAL_HEADERS).
--define(LOW_BALANCE_VALUES, [{<<"Event-Category">>, <<"notification">>}
-                            ,{<<"Event-Name">>, <<"low_balance">>}
-                            ]).
--define(LOW_BALANCE_TYPES, []).
+%%%===================================================================
+%%% Internal Notifications Definitions
+%%%===================================================================
+
+
+%% Notify Update
+notify_update_definition() ->
+    #kapi_definition{name = <<"notify_update">>
+                    ,friendly_name = <<"Notify Status Update">>
+                    ,description = <<"This event is triggered by notification consumer, e.g. teletype application, to send status of notification being process to publisher">>
+                    ,build_fun = fun notify_update/1
+                    ,validate_fun = fun notify_update_v/1
+                    ,publish_fun = fun publish_notify_update/2
```

Comment:
Spec'd as being an arity-1 function. Update spec to account for arity-2 functions, or ?

Reply:
a quick look at all API publish functions, we have so many functions with different arity (range from 0 to 6) with different args order. So I changed it to just have_`fun()` now.